### PR TITLE
SASS-8094: Added the scalafmt formatter to the repository

### PR DESCRIPTION
**Pull request by vivekhmrc**

### CODE REVIEW

File: .scalafmt [hunk 1]
```diff
@@ -0,0 +1,51 @@
+version = 3.5.9
```
Comment (arunkumar461) [inline .scalafmt:1]: file name should be .scalafmt.conf

File: .scalafmt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+runner.dialect = scala213
```
Comment (arunkumar461) [inline .scalafmt:2]: Can we also refer HMRC other repos for patterns?
https://github.com/hmrc/play-frontend-hmrc/blob/main/.scalafmt.conf

**Review by arunkumar461 — CHANGES_REQUESTED** [review]

File: .scalafmt.conf [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  beforeContextBoundColon = false
+}
+rewriteTokens {
+  "\t": "  "
+  "→": "->"
+  "←": "<-"
+  "⇒": "=>"
+}
+
+unindentTopLevelOperators = true
```
Comment (arunkumar461) [inline .scalafmt.conf:47]: image

**Review by arunkumar461 — CHANGES_REQUESTED** [review]

File: .scalafmt.conf (hunk 3 above)
Comment (vivekhmrc) [reply .scalafmt.conf]: Valid value for an older version.

**Review by arunkumar461 — APPROVED** [review]

**Pull request merged**